packages/xml-lsp-server/src/server.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  TextDocumentChangeEvent,
+  Diagnostic,
+  InitializeParams,
+  DidChangeConfigurationNotification
+} from "vscode-languageserver";
+import { configureSettings, validateDocument } from "./serverHelper";
+
+const diagnosticsCache = new Map<string, Map<string, Diagnostic[]>>();
+const settings = configureSettings();
+const connection = settings.connection;
+const documents = settings.documents;
+
+let hasConfigurationCapability = false;
+let hasWorkspaceFolderCapability = false;
+
+connection.onInitialize((params: InitializeParams) => {
+  const capabilities = params.capabilities;
+
+  // Does the client support the `workspace/configuration` request?
+  // If not, we will fall back using global settings
+  hasConfigurationCapability = !!(
+    capabilities.workspace && !!capabilities.workspace.configuration
+  );
+  hasWorkspaceFolderCapability = !!(
+    capabilities.workspace && !!capabilities.workspace.workspaceFolders
+  );
+
+  return {
+    capabilities: {
+      textDocumentSync: documents.syncKind
+    }
+  };
+});
+
+connection.onInitialized(() => {
+  if (hasConfigurationCapability) {
+    // Register for all configuration changes
+    connection.client.register(
+      DidChangeConfigurationNotification.type,
+      undefined
+    );
+  }
+  if (hasWorkspaceFolderCapability) {
```

Comment:
if we are not doing anything with this event currently, do we need the event handler?